### PR TITLE
Add UserSettings on application context.

### PR DESCRIPTION
**Pull request by thibaultzanini**
Ticket: #17670

### CODE REVIEW

**Comment by thibaultzanini** [comment]
PR updated

File: core/service/application-context-service.js [hunk 1]
```diff
@@ -119,24 +119,28 @@ var ApplicationContextService = exports.ApplicationContextService = Montage.spec
...
                 }
 
-                if (!applicationRawContext.userSettings) {
+                if (!applicationRawContext.userSettings && typeof applicationRawContext.userSettings !== "object") {
                     applicationRawContext.userSettings = {};
                 }
 
-                applicationRawContext.dashboardContext.widgets = applicationRawContext.dashboardContext.widgets || [];
-                applicationRawContext.sideBoardContext.widgets = applicationRawContext.sideBoardContext.widgets || [];
+                if (!applicationRawContext.dashboardContext.widgets && !Array.isArray(applicationRawContext.dashboardContext.widgets)) {
```
Comment (pchaussalet) [inline core/service/application-context-service.js]: Shouldn't this be a OR (||) condition ?

File: core/service/application-context-service.js [hunk 2]
```diff
@@ -119,24 +119,28 @@ var ApplicationContextService = exports.ApplicationContextService = Montage.spec
...
+                    applicationRawContext.dashboardContext.widgets = [];
+                }
+
+                if (!applicationRawContext.sideBoardContext.widgets && !Array.isArray(applicationRawContext.sideBoardContext.widgets)) {
```
Comment (pchaussalet) [inline core/service/application-context-service.js]: Shouldn't this be a OR (||) condition ?

File: core/service/application-context-service.js [hunk 3]
```diff
@@ -119,24 +119,28 @@ var ApplicationContextService = exports.ApplicationContextService = Montage.spec
...
     },
 
-    _handleRawApplicationContext: {
+    _repairApplicationRawContextIfNeeded: {
         value: function (applicationRawContext) {
-            if (applicationRawContext) {
-                // Repair applicationRawContext if needed
-                if (!applicationRawContext.dashboardContext) {
+            if (applicationRawContext) {                
+                if (!applicationRawContext.dashboardContext && typeof applicationRawContext.dashboardContext !== "object") {
```
Comment (pchaussalet) [inline core/service/application-context-service.js]: Shouldn't this be a OR (||) condition ?

File: core/service/application-context-service.js [hunk 4]
```diff
@@ -119,24 +119,28 @@ var ApplicationContextService = exports.ApplicationContextService = Montage.spec
...
                     applicationRawContext.dashboardContext = {};
                 }
 
-                if (!applicationRawContext.sideBoardContext) {
+                if (!applicationRawContext.sideBoardContext && typeof applicationRawContext.sideBoardContext !== "object") {
```
Comment (pchaussalet) [inline core/service/application-context-service.js]: Shouldn't this be a OR (||) condition ?

File: core/service/application-context-service.js [hunk 5]
```diff
@@ -119,24 +119,28 @@ var ApplicationContextService = exports.ApplicationContextService = Montage.spec
...
                     applicationRawContext.sideBoardContext = {};
                 }
 
-                if (!applicationRawContext.userSettings) {
+                if (!applicationRawContext.userSettings && typeof applicationRawContext.userSettings !== "object") {
```
Comment (pchaussalet) [inline core/service/application-context-service.js]: Shouldn't this be a OR (||) condition ?

**Review by pchaussalet — CHANGES_REQUESTED** [review]

**Review by pchaussalet — APPROVED** [review]

**Pull request merged**